commands/channel.go

```diff
@@ -479,3 +495,28 @@ func searchChannelCmdF(c client.Client, cmd *cobra.Command, args []string) error
 	}
 	return nil
 }
+
+func moveChannelCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	team := getTeamFromTeamArg(c, args[0])
+	if team == nil {
+		return fmt.Errorf("unable to find destination team %q", args[0])
+	}
+
+	removeDeactivatedMembers, _ := cmd.Flags().GetBool("remove-deactivated-users")
+
+	channels := getChannelsFromChannelArgs(c, args[1:])
+	for i, channel := range channels {
+		if channel == nil {
+			printer.PrintError(fmt.Sprintf("Unable to find channel %q", args[i+1]))
+			continue
+		}
+
+		if _, resp := c.MoveChannel(channel.Id, team.Id, removeDeactivatedMembers); resp.Error != nil {
+			printer.PrintError(fmt.Sprintf("unable to move channel %q: %s", channel.Name, resp.Error))
+			continue
+		}
+		originTeamID := channel.TeamId
+		printer.Print(fmt.Sprintf("Moved channel %q to %q (%s) from %s.", channel.Name, team.Name, team.Id, originTeamID))
```

Comment:
We should capture the new channel (`newChannel` in the following suggestion) and print it in case the user passes the `--format json` flag:

```suggestion
		printer.PrintT(fmt.Sprintf("Moved channel {{.Name}} to %q ({{.TeamId}}) from %s.", team.Name, originTeamId), newChannel)
```